apps/dotcom/client/src/tla/components/TlaSidebar/TlaSidebar.tsx

```diff
@@ -215,8 +217,12 @@ function TlaSidebarFileLink({ item }: { item: RecentFile }) {
 	const { fileSlug } = useParams()
 	const isActive = TldrawAppFileRecordType.createId(fileSlug) === fileId
 	const [isRenaming, setIsRenaming] = useState(false)
+	const trackEvent = useTldrawAppUiEvents()
 
-	const handleRenameAction = () => setIsRenaming(true)
+	const handleRenameAction = () => {
+		setIsRenaming(true)
+		trackEvent('rename-file', { source: 'sidebar' })
```

Comment:
Similar thing here, we probably need to move this to `TlaRenameInline`.

Reply:
Putting it in the handleSave function in the TlaRenameInline component, so it doesn't fire every keystroke.